pkg/builder/interface.go

```diff
@@ -92,8 +92,7 @@ func IsTimeout(status *v1alpha1.BuildStatus, buildTimeout string) bool {
 			return false
 		}
 	}
-
-	return time.Since(status.StartTime.Time).Seconds() > timeout.Seconds()
+	return time.Since(status.CreationTime.Time).Seconds() > timeout.Seconds()
```

Comment:
Time spent scheduling the pod shouldn't count toward the build's timeout, should this consider `status.StartTime` instead of `status.CreationTime` ?

Unless we were thinking this would be a useful way to kill a build that was unschedulable, in which case we might need to introduce two separate timeouts.

In GCB we only enforce timeouts once the build starts, since it's not the user's fault if it takes us a long time to start their build. We've considered enforcing a timeout on queued time, but I don't think any user has asked for it.

Reply:
If we don't enforce timeout for allocating resources then build will remain in `pending` state forever. I was planning to follow up with a PR to introduce another timeout for allocating resources(queue timeout). Probably default this timeout to 10min and user doesn't need to worry about using this.

Reply:
Okay, if that's on your radar for future changes that sounds good to me.

Reply:
Created: https://github.com/knative/build/issues/325 to follow up with changes discussed in this PR.